MediaManager/Queue/MediaQueue.cs

```diff
@@ -36,6 +64,17 @@ public int CurrentIndex
             }
         }
 
+        public void Shuffle()
```

Comment:
This is probably not gonna work. Shuffle needs to be able to be undone as well. Not sure about optimal solution, maybe keeping a double list, but that gave problems in the past, or maybe a custom getter to return only a shuffled list temporary.